File: less_to_json.pl
use strict;
use warnings;

my @data;

open my $fh, "<:encoding(UTF-8)", "open-color.less" or die $!;
while (<$fh>) {
	next unless /^\@oc-([a-z]+)-list:\s*([^;]+);?$/;
	push @data, (
		sprintf '%10s: [%s]', qq/"$1"/, (
			join ', ', map{qq/"$_"/} split(/\s*,\s*/, $2)
		)
	)
}
close $fh;

open my $fs, ">:encoding(UTF-8)", "open-color.json" or die $!;
print $fs "{\n" . (join ",\n", @data) . "\n}\n";
close $fs;

File: less_to_json.js
const {readFileSync, writeFileSync} = require('fs')

const data = []

// see also: https://www.npmjs.com/package/left-pad
String.prototype.leftPad = function(len) {
	return ' '.repeat(len - this.length) + this
}

readFileSync('open-color.less').toString('utf-8').split(/\r?\n/).forEach(line => {
	var m
	if((m = line.match(/^@oc-([a-z]+)-list:\s*([^;]+);?$/)) == null) return;
	const key = (`"${m[1]}"`).leftPad(10)
	const value = m[2].split(/\s*,\s*/).map(v=>`"${v}"`).join(', ')
	data.push(`${key}: [${value}]`)
})

writeFileSync('open-color.json',
	['{', data.join(',\n'), '}'].join('\n') + '\n'
)
